Separate c10::Symbol header from list of interned strings

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #69407
* __->__ #69406

Most files that include `interned_strings.h` don't actually depend on
anything generated from `FORALL_NS_SYMBOLS` yet because they're in a
single file you need to recompile whenever a new symbol is added. Here
I move the class definition into a separate file so this doesn't
happen.

Differential Revision: [D32923637](https://our.internmc.facebook.com/intern/diff/D32923637)

peterbell10 committed:
Separate c10::Symbol header from list of interned strings

Most files that include `interned_strings.h` don't actually depend on
anything generated from `FORALL_NS_SYMBOLS` yet because they're in a
single file you need to recompile whenever a new symbol is added. Here
I move the class definition into a separate file so this doesn't
happen.

[ghstack-poisoned]

peterbell10 committed:
Update on "Separate c10::Symbol header from list of interned strings"


Most files that include `interned_strings.h` don't actually depend on
anything generated from `FORALL_NS_SYMBOLS` yet because they're in a
single file you need to recompile whenever a new symbol is added. Here
I move the class definition into a separate file so this doesn't
happen.

[ghstack-poisoned]

peterbell10 committed:
Update on "Separate c10::Symbol header from list of interned strings"


Most files that include `interned_strings.h` don't actually depend on
anything generated from `FORALL_NS_SYMBOLS` yet because they're in a
single file you need to recompile whenever a new symbol is added. Here
I move the class definition into a separate file so this doesn't
happen.

[ghstack-poisoned]

albanD reviewed (approved):
Thanks!

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32923637).

peterbell10 committed:
Update on "Separate c10::Symbol header from list of interned strings"


Most files that include `interned_strings.h` don't actually depend on
anything generated from `FORALL_NS_SYMBOLS` yet because they're in a
single file you need to recompile whenever a new symbol is added. Here
I move the class definition into a separate file so this doesn't
happen.

Differential Revision: [D32923637](https://our.internmc.facebook.com/intern/diff/D32923637)

[ghstack-poisoned]

albanD commented:
@albanD has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32923637).
